src/App/Components/CreateArticle/UpdateArticle.js

```diff
@@ -0,0 +1,385 @@
+/* eslint-disable no-shadow */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { Redirect } from 'react-router-dom';
+import propTypes from 'prop-types';
+import jwt from 'jwt-decode';
+
+import SimplexEditor from 'simple-react-editor';
+import Tagify from 'react-tagify-section';
+import queryString from 'query-string';
+import Navbar from '../Common/NavProfile/navbar';
+import Footer from '../Common/Footer';
+import Loader from '../Common/loader';
+import {
+  categories as getCategories,
+  getTags,
+  changeState,
+  drafting,
+  publish
+} from '../../../Redux/Actions/articles';
+import updateArticle from '../../../Redux/Actions/getArticle';
+import imageUploader from '../../../Helpers/image-upload';
+import Selector from './Selector';
+import NotFound from '../ArticleNotFound/ArticleNotFound';
+
+const { getArticleDetail, getArticleTags, getDraftedArticle } = updateArticle;
+export class UpdateArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      article: null,
+      categories: [],
+      tags: [],
+      imgUrl: null,
+      isCoverImage: false,
+      stateTags: [],
+      initialState: null,
+      published: false,
+      isLoading: true,
+      title: '',
+      description: '',
+      isChanged: false
+    };
+
+    /* istanbul ignore next */
+    this.autosaveFunctionality = setInterval(() => {
+      const {
+        stateTags, article, imgUrl, isChanged
+      } = this.state;
+      const { drafting: autosaveArticle, match } = this.props;
+      const { slug } = match.params;
+      const data = {
+        ...article, slug, tags: stateTags, coverImage: imgUrl
+      };
+      if (isChanged) {
+        autosaveArticle(data);
+        this.setState({ isChanged: false });
+      }
+    }, 6000);
+  }
+
+  componentWillMount() {
+    const {
+      getCategories,
+      getArticleTags,
+      getDraftedArticle,
+      location,
+      match,
+      getArticleDetail,
+      getTags
+    } = this.props;
+
+    getCategories();
+    getTags();
+    const { draft } = queryString.parse(location.search);
+    const { slug } = match.params;
+
+    if (draft === 'true') {
+      getDraftedArticle(slug);
+      getArticleTags(slug);
+      setTimeout(() => {
```

Comment:
@truestbyheart why do you need a `setTimeout` here?

Reply:
Use this to allow the loader to show up and the data to be properly set up

Reply:
my concern with using a timeout is what if it (i.e what you are waiting for) takes longer than the time you set? Remember, not all your users will be using your kind of internet or device. I would opt for a more dependable option.

Reply:
the setTimeout is wrapped in an if-else statement meaning in each situation the setTimeout will finish and the proper actions will be taken...this simulate that we may have the data or not but we show our client we tried processing the request